storage/merkle/tree.go

```diff
@@ -348,76 +407,41 @@ DelLoop:
 	// child's child as the child of the merged short node
 	c, ok := n.child.(*short)
 	if ok {
-		t.merge(n, c)
+		merge(n, c)
 	}
 
 	// if the parent is also a short node, we have to merge them and use the
 	// current child as the child of the merged node
 	p, ok := (*parent).(*short)
 	if ok {
-		t.merge(p, n)
+		merge(p, n)
 	}
 
 	// NOTE: if neither the parent nor the child are short nodes, we simply
 	// bypass both conditional scopes and land here right away
-
 	return true
 }
 
-// Hash will return the root hash of this patricia merkle tree.
+// Hash returns the root hash of this patricia merkle tree.
 func (t *Tree) Hash() []byte {
-	hash := t.nodeHash(t.root)
-	return hash
+	if t.size == 0 {
+		return []byte{14, 87, 81, 192, 38, 229, 67, 178, 232, 171, 46, 176, 96, 153, 218, 161, 209, 229, 223, 71, 119, 143, 119, 135, 250, 171, 69, 205, 241, 47, 227, 168}
+	}
+	return t.root.Hash()
 }
 
 // merge will merge a child short node into a parent short node.
-func (t *Tree) merge(p *short, c *short) {
+func merge(p *short, c *short) {
 	totalCount := p.count + c.count
-	totalPath := bitset.NewBytes(int(totalCount))
-	parentPath := bitset.Bytes(p.path)
-	for i := uint(0); i < p.count; i++ {
-		totalPath.SetBool(int(i), parentPath.Get(int(i)))
+	totalPath := bitutils.MakeBitVector(totalCount)
+	for i := 0; i < p.count; i++ {
+		bitutils.SetBit(totalPath, i, bitutils.Bit(p.path, i))
 	}
-	childPath := bitset.Bytes(c.path)
-	for i := uint(0); i < c.count; i++ {
-		totalPath.SetBool(int(i+p.count), childPath.Get(int(i)))
+	//childPath := bitset.Bytes(c.path)
```

Comment:
I guess we do not need the commented line anymore?
```suggestion
```

Reply:
included in https://github.com/onflow/flow-go/pull/1800/commits/050a98873836fb6f63e566a9668927232741a7ac